pkg/services/ngalert/store/alert_rule.go

```diff
@@ -503,6 +503,10 @@ func (st DBstore) validateAlertRule(alertRule ngmodels.AlertRule) error {
 		return fmt.Errorf("%w: no organisation is found", ngmodels.ErrAlertRuleFailedValidation)
 	}
 
+	if alertRule.DashboardUID == nil && alertRule.PanelID != nil {
+		return errors.New("cannot have Panel ID without a Dashboard UID")
```

Comment:
```suggestion
		return errors.New("%w: cannot have Panel ID without a Dashboard UID", ngmodels.ErrAlertRuleFailedValidation)
```

The API code asserts on `ngmodels.ErrAlertRuleFailedValidation` as it unwraps the error - this should fix the 400 instead of the 500.

https://github.com/grafana/grafana/blob/71c6d7285e8dffe68195c9259ef62b04c4346fb8/pkg/services/ngalert/api/api_ruler.go#L280-L281